### PR TITLE
Ss/dcos 49721

### PR DESCRIPTION
**Pull request by SarahStegall**
## Description
https://jira.mesosphere.com/browse/DCOS-49721

The CLI Command Reference pages are formatted in many different styles. In particular, some pages such as the `dcos security` pages are formatted as shell scripts throughout. The Command Reference files need to be formatted in the same style throughout. This has to be completed before the docs are sent to be translated into Chinese, because otherwise none of the options, flags, or other variables would be translated.

## Urgency
- [X] Blocker 
- [ ] High
- [ ] Medium

### CODE REVIEW

**Comment by SarahStegall** [comment]
The following commands seem to be missing or have errors, based on the output from commands issued on the CLI (running Enterprise CLI for 1.12).

dcos node
	dcos node dns
	dcos node metrics details
	dcos node metrics summary

dcos job
  	dcos job queue

dcos license
	dcos audit get (something is wrong with this)

dcos task
	dcos task kill

dcos package
	dcos package update is no longer in dcos but is in docs

dcos task
	dcos task metrics details
	dcos task metrics summary

**Comment by jpe442** [comment]
> The following commands seem to be missing or have errors, based on the output from commands issued on the CLI (running Enterprise CLI for 1.12).
> 
> dcos node
> 
> * dcos node dns
> * dcos node metrics details
> * dcos node metrics summary
> 
> dcos job
> 
> * dcos job queue
> 
> dcos license
> 
> * dcos audit get (something is wrong with this)
> 
> dcos task
> 
> * dcos task kill
> 
> dcos package
> 
> * dcos package update (no longer in `dcos` but is in docs)
> 
> dcos task
> 
> * dcos task metrics details
> * dcos task metrics summary

@SarahStegall --- Can we open a ticket and better describe these potential short-comings?

**Comment by SarahStegall** [comment]
Since all of these edits are in support of eventually sending these docs to be translated into Chinese, I left them under this JIRA ticket.

File: pages/1.12/cli/command-reference/dcos-help/index.md [hunk 1]
```diff
@@ -16,19 +16,92 @@ The `dcos help` command will display DC/OS CLI help information.
 ```bash
 dcos help <subcommand>
 ```
+Output of `dcos help` (without options) should look like this:
```
Comment (rgo3) [inline pages/1.12/cli/command-reference/dcos-help/index.md]: This help output seems incorrect for the 1.12 (0.7.x) CLI.
It should look something like this:
```
Usage:
  dcos [command]

Commands:
  auth
      Authenticate to DC/OS cluster
  cluster
      Manage your DC/OS clusters
  config
      Manage the DC/OS configuration file
  help
      Help about any command
  job
      Deploy and manage jobs in DC/OS
  marathon
      Deploy and manage applications to DC/OS
  node
      View DC/OS node information
  package
      Install and manage DC/OS software packages
  plugin
      Manage CLI plugins
  service
      Manage DC/OS services
  task
      Manage DC/OS tasks

Options:
  --version
      Print version information
  -v, -vv
      Output verbosity (verbose or very verbose)
  -h, --help
      Show usage help

Use "dcos [command] --help" for more information about a command.
```

File: pages/1.12/cli/index.md [hunk 2]
```diff
@@ -10,49 +10,63 @@ enterprise: false
...
 
-DC/OS 1.12 requires the DC/OS CLI >= 0.7. To install it, [follow the instructions](/1.12/cli/install/).
+DC/OS 1.12 requires DC/OS CLI >= 0.7. To install it, [follow the instructions](/1.12/cli/install/).
 
 To list available commands, run `dcos` with no parameters:
 
 ```bash
 $ dcos
 
-Usage:
```
Comment (rgo3) [inline pages/1.12/cli/index.md]: This seems to be the correct output if you run `dcos`.

File: pages/1.12/cli/command-reference/dcos/index.md [hunk 3]
```diff
@@ -20,22 +22,29 @@ dcos [options] [<command>] [<args>...]
...
 | `--help, h`   |  Display usage. |
-| `--log-level=<log-level>`  | Set the logging level. This setting does not affect the output sent to `stdout`.  |
-|  `--version, v`  |  Display version information.  |
-| `<log-level>` | The severity levels are shown in Table 2. |
+|  `--debug`  |  Enable debug mode. |
+| `--version` | Display version information. |
+| `--log-level=<log-level>`  | Set the logging level. This setting does not affect the output sent to `stdout`.  See below for severity levels. |
+
+
+## Log level severity
```
Comment (rgo3) [inline pages/1.12/cli/command-reference/dcos/index.md]: This seems outdated. The 1.12 CLI uses `-v` and `-vv` for verbosity of logging. The `--log-level` flag still works but was deprecated with 1.12.

File: pages/1.12/cli/command-reference/dcos-security/dcos-security-org/dcos-security-org-groups-create/index.md [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+enterprise: true
+---
+# Description
+The `dcos security org groups create` command allows you to create a new group.
+
+# Options
+
+| Name |  Description |
+|---------|-------------|
+| `d`, `--description <text>` | Description of group.|
```
Comment (jongiddy) [inline pages/1.12/cli/command-reference/dcos-security/dcos-security-org/dcos-security-org-groups-create/index.md]: Should be `-d`

File: pages/1.12/cli/command-reference/dcos-security/dcos-security-cluster/dcos-security-cluster-directory-import-group/index.md [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+navigationTitle:  dcos security cluster directory import_group
+title: dcos security cluster directory import_group
+menuWeight: 33
+excerpt: Importing an LDAP group
+enterprise: true
+---
+
+# Description
+
+The `dcos security cluster directory import_group` command imports a group of users from the configured directory (LDAP) backend. See IAM documentation for details on group import.
```
Comment (jongiddy) [inline pages/1.12/cli/command-reference/dcos-security/dcos-security-cluster/dcos-security-cluster-directory-import-group/index.md]: Last sentence can link to `/1.12/security/ent/users-groups/add-external-user/#importing-groups-of-ldap-users`

File: pages/1.12/cli/command-reference/dcos-security/dcos-security-cluster/dcos-security-cluster-secret-store-seal-status/index.md [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+
+```
+dcos security cluster secret-store seal-status [OPTIONS] STORE_ID
+```
+
+# Options
+
+| Name |  Description |
+|---------|-------------|
+| `STORE_ID`  | ID of secret store. (Required)|
```
Comment (jongiddy) [inline pages/1.12/cli/command-reference/dcos-security/dcos-security-cluster/dcos-security-cluster-secret-store-seal-status/index.md]: In `dcos auth list-providers` command line arguments of this form are listed under "Positional arguments".

File: pages/1.12/cli/command-reference/dcos-security/dcos-security-cluster/dcos-security-cluster-oidc-modify/index.md [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+navigationTitle:  cos security cluster oidc modify
+title: cos security cluster oidc modify
+menuWeight: 60
+excerpt: Modifying an existing OIDC provider configuration
+enterprise: true
+---
+
+# Description
+
+The `cos security cluster oidc modify` command lets you modify an existing OIDC provider configuration.
```
Comment (jongiddy) [inline pages/1.12/cli/command-reference/dcos-security/dcos-security-cluster/dcos-security-cluster-oidc-modify/index.md]: `cos` -> `dcos` here and twice in the metadata above.

File: pages/1.12/cli/command-reference/dcos-security/dcos-security-cluster/dcos-security-cluster-secret-store-unseal/index.md [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+
+```
+dcos security cluster secret-store unseal [OPTIONS] STORE_ID KEY
+```
+
+# Options
+
+| Name |  Description |
+|---------|-------------|
+| `STORE_ID KEY`  | ID key of secret store. (Required) |
```
Comment (jongiddy) [inline pages/1.12/cli/command-reference/dcos-security/dcos-security-cluster/dcos-security-cluster-secret-store-unseal/index.md]: These are two distinct parameters. They should be placed on separate lines. 

@mhrabovcin What are good descriptions for these parameters?

File: pages/1.12/cli/command-reference/dcos-help/index.md (hunk 1 above)
Comment (SarahStegall) [reply pages/1.12/cli/command-reference/dcos-help/index.md]: Further investigation:
I spun up a 1.12 cluster and loaded the CLI according to instructions. Here is the info I got from `dcos --version`:

dcos --version
dcoscli.version=0.6.1
dcos.version=1.12.0
dcos.commit=20fa047bbd37188ccb55f61ab9590edc809030ec
dcos.bootstrap-id=51c3e1a0a5b7fdace4abccc998e970df07eb606e`

You will see that even though this is 1.12, I am still only able to download 0.6.1. Apparently this is the only version available to 1.12 and 1.13. Even the upgrade/patching instructions do not tell users how to select a specific version of the CLI.

**How did you get 0.7.1 loaded?**

File: pages/1.12/cli/command-reference/dcos-auth/dcos-auth-logout/index.md [hunk 9]
```diff
@@ -3,14 +3,14 @@ layout: layout.pug
 navigationTitle:  dcos auth logout
 title: dcos auth logout
 menuWeight: 3
-excerpt: Logging out of DC/OS authentication
+excerpt: Logging out of a DC/OS cluster
```
[the remark this answers is not in the document]
Comment (SarahStegall) [reply pages/1.12/cli/command-reference/dcos-auth/dcos-auth-logout/index.md]: Good catch, guys! I will fix these inconsistencies.

File: pages/1.12/cli/command-reference/dcos-auth/dcos-auth-list-providers/index.md [hunk 10]
```diff
@@ -9,25 +9,27 @@ enterprise: true
 ---
 
 # Description
-This command lists configured authentication providers for your DC/OS cluster. For more information, see [Service Accounts](/1.12/security/ent/service-auth/).
+
+The `dcos auth list-providers` command lists configured authentication providers for your DC/OS cluster. For more information, see [Service Accounts](/1.12/security/ent/service-auth/).
```
[the remark this answers is not in the document]
Comment (SarahStegall) [reply pages/1.12/cli/command-reference/dcos-auth/dcos-auth-list-providers/index.md]: The output of `dcos auth list-providers` is:

`dcos auth list-providers
PROVIDER ID    AUTHENTICATION TYPE                                                   
dcos-services  Log in using a DC/OS service user account (username and private key)  
dcos-users     Log in using a standard DC/OS user account (username and password)    `

I was thinking that a user who was told they need a service user account would need a link to the page telling him how to acquire one.

File: pages/1.12/cli/command-reference/dcos-auth/dcos-auth-login/index.md [hunk 11]
```diff
@@ -3,30 +3,35 @@ layout: layout.pug
...
-|  `--private-key=<key_path>`  |         | The path to a file that contains the private key.  |
-|  `--provider=<provider_id>`  |         | [enterprise type="inline" size="small" /] The authentication provider to use for login.  |
-|  `--username=<username>`  |            | The username for login. |
+| `--help, h`   | Display usage. |
+| `--info`   | Display a short description of this subcommand. |
+| `--version`   | Display version information. |
+|  `--password-file=<password_file>`  |  The path to a file that contains the password for login. |
+|  `--password=<password>`  |     Specify the password for login on the command line. This is insecure.  |
+|  `--private-key=<key_path>`  |   The path to a file that contains the private key.  |
+|  `--provider=<provider_id>`  |   [enterprise type="inline" size="small" /] The authentication provider to use for login.  |
```
[the remark this answers is not in the document]
Comment (SarahStegall) [reply pages/1.12/cli/command-reference/dcos-auth/dcos-auth-login/index.md]: Not sure what you mean in the first comment by "valid". Was it not valid before? Also, are we still talking about DC/OS 1.12?

File: pages/1.12/cli/command-reference/dcos-security/index.md [hunk 12]
```diff
@@ -7,614 +7,15 @@ excerpt: Managing the DC/OS Certificate Authority
 
 enterprise: true
 ---
-The `dcos security` command allows you to manage your DC/OS certificate authority credentials.
+The `dcos security` commands allow you to manage your DC/OS cluster security.
```
[the remark this answers is not in the document]
Comment (SarahStegall) [reply pages/1.12/cli/command-reference/dcos-security/index.md]: All of the commands that are available exclusively via DC/OS Enterprise are tagged with a purple "Enterprise" button on each page.

File: pages/1.12/cli/command-reference/dcos-security/dcos-security-cluster/dcos-security-cluster-directory-import-group/index.md (hunk 5 above)
Comment (SarahStegall) [reply pages/1.12/cli/command-reference/dcos-security/dcos-security-cluster/dcos-security-cluster-directory-import-group/index.md]: Will do. Thanks!

File: pages/1.12/cli/command-reference/dcos-security/dcos-security-cluster/dcos-security-cluster-oidc-modify/index.md (hunk 7 above)
Comment (SarahStegall) [reply pages/1.12/cli/command-reference/dcos-security/dcos-security-cluster/dcos-security-cluster-oidc-modify/index.md]: Wow! Thanks for catching that!

File: pages/1.12/cli/command-reference/dcos-security/dcos-security-org/dcos-security-org-groups-create/index.md (hunk 4 above)
Comment (SarahStegall) [reply pages/1.12/cli/command-reference/dcos-security/dcos-security-org/dcos-security-org-groups-create/index.md]: Got it.

File: pages/1.12/cli/command-reference/dcos-security/dcos-security-cluster/dcos-security-cluster-secret-store-seal-status/index.md (hunk 6 above)
Comment (SarahStegall) [reply pages/1.12/cli/command-reference/dcos-security/dcos-security-cluster/dcos-security-cluster-secret-store-seal-status/index.md]: Ok, thanks. Here's what I got from the help command:

`Usage: dcos security cluster secret-store seal-status [OPTIONS] STORE_ID`

The options only listed --json and --help, and there are no positional arguments listed in the CLI output. I had to guess. :) I will fix this as noted.

File: pages/1.12/cli/command-reference/dcos-security/dcos-security-cluster/dcos-security-cluster-secret-store-unseal/index.md (hunk 8 above)
Comment (SarahStegall) [reply pages/1.12/cli/command-reference/dcos-security/dcos-security-cluster/dcos-security-cluster-secret-store-unseal/index.md]: Per your comment above, I will list this as a Positional Argument.

File: pages/1.12/cli/command-reference/dcos-auth/dcos-auth-login/index.md (hunk 11 above)
[the remark this answers is not in the document]
Comment (jongiddy) [reply pages/1.12/cli/command-reference/dcos-auth/dcos-auth-login/index.md]: By "valid in Open", I mean that it is not Enterprise-only, but you are correct that change is for 1.13. So, here it should still have the Enterprise label.

File: pages/1.12/cli/command-reference/dcos-auth/dcos-auth-list-providers/index.md (hunk 10 above)
[the remark this answers is not in the document]
Comment (jongiddy) [reply pages/1.12/cli/command-reference/dcos-auth/dcos-auth-list-providers/index.md]: It's possible that we should have a table of all the possible providers with a brief description and even a link to more information.

But at the moment, this link seems to focus attention on one provider, which is probably not what the caller is looking for. Service accounts are always supported in Enterprise, so the caller of this command is probably looking to identify whether LDAP, OpenID Connect, or SAML are supported. The pages `/1.12/security/ent/ldap/` and `/1.12/security/ent/sso/` would be at least as relevant.

**Pull request merged**